src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -762,31 +790,282 @@ public void Dispose()
+        /// The class contains information about the source, the destination and ping results.
+        /// </summary>
+        public class PingMtuStatus : PingStatus
+        {
+            /// <summary>
+            /// Initializes a new instance of the <see cref="PingMtuStatus"/> class.
+            /// </summary>
+            /// <param name="source">The source machine name or IP of the ping.</param>
+            /// <param name="destination">The destination machine name of the ping.</param>
+            /// <param name="reply">The response from the ping attempt.</param>
+            /// <param name="bufferSize">The buffer size from the successful ping attempt.</param>
+            internal PingMtuStatus(string source, string destination, PingReply reply, int bufferSize)
+                : base(source, destination, reply, 1)
+            {
+                MtuSize = bufferSize;
+            }
+
+            /// <summary>
+            /// Gets the maximum transmission unit size on the network path between the source and destination.
+            /// </summary>
+            public int MtuSize { get; }
+        }
+
+        /// <summary>
+        /// The class contains an information about a trace route attempt.
+        /// </summary>
+        public class TraceStatus
+        {
+            /// <summary>
+            /// Initializes a new instance of the <see cref="TraceStatus"/> class.
+            /// </summary>
+            /// <param name="hop">The hop number of this trace hop.</param>
+            /// <param name="status">The PingStatus response from this trace hop.</param>
+            /// <param name="source">The source computer name or IP address of the traceroute.</param>
+            /// <param name="destination">The target destination of the traceroute.</param>
+            /// <param name="destinationAddress">The target IPAddress of the overall traceroute.</param>
+            internal TraceStatus(
+                int hop,
+                PingStatus status,
+                string source,
+                string destination,
+                IPAddress destinationAddress)
+            {
+                _status = status;
+                Hop = hop;
+                Source = source;
+                Target = destination;
+                TargetAddress = destinationAddress;
+            }
+
+            private readonly PingStatus _status;
+
+            /// <summary>
+            /// Gets the number of the current hop / router.
+            /// </summary>
+            public int Hop { get; }
+
+            /// <summary>
+            /// Gets the hostname of the current hop point.
+            /// </summary>
+            /// <value></value>
+            public string Hostname
+            {
+                get => _status.Destination != IPAddress.Any.ToString()
+                    && _status.Destination != IPAddress.IPv6Any.ToString()
+                        ? _status.Destination
+                        : null;
```

Comment:
Why do we need return null? Why not  _status.Destination?

Reply:
If a `PingReply` comes back in an error state, we get either `0.0.0.0` or `::` (referenced by the `Any` addresses in the code here) depending if it's IPv4 or IPv6. In both cases this is not especially meaningful to the user. I think `ping` utils typically indicated missing data with `*` as I have in the formatter for other cases.

I have tried to keep such formatting concerns out of the objects themselves; I think it would be simpler for users to interact with the objects if the value is simply `null` rather than some string on the actual object.

Reply:
If the type is exclusively for displaying what about DisplayHostname ?

Reply:
Perhaps. I'm not sure it makes sense to double up on all the properties, though, and although scenarios may not be incredibly common, I would _like_ for the output data to still be easily usable programmatically.

Reply:
That said, you're right in that it might be a good idea to just have the formatter reference display properties... 🤷‍♂ Not sure if that's better or about the same as the formatter doing it with scriptblocks to be honest.